crucible-syntax/src/Lang/Crucible/Syntax/Concrete.hs

```diff
@@ -1749,49 +1842,53 @@ termStmt' retTy =
            -> m (Ctx.Assignment (Atom s) argTypes)
         go Ctx.AssignEmpty = emptyList *> pure Ctx.empty
         go (Ctx.AssignExtend tys ty) =
-          depCons (located (reading (check ty))) $
+          depCons (located (reading (check ty hooks))) $
             \(Posd loc arg) ->
                Ctx.extend <$> go (Ctx.viewAssign tys) <*> eval loc arg
 
     err :: m (TermStmt s ret)
     err =
-      do Posd loc e <- unary Error_ (located (reading (check (StringRepr UnicodeRepr))))
+      do Posd loc e <- unary Error_ (located (reading (check (StringRepr UnicodeRepr) hooks)))
          ErrorStmt <$> eval loc e
 
     out :: m (TermStmt s ret)
     out = followedBy (kw Output_) $
           do -- commit
              depCons lambdaLabel $
                \(Pair argTy lbl) ->
-                 depCons (located (reading (check argTy))) $
+                 depCons (located (reading (check argTy hooks))) $
                    \(Posd loc arg) ->
                      emptyList *>
                        (Output lbl <$> eval loc arg)
 
 
 
-data Rand s t = Rand (AST s) (E s t)
+data Rand s ext t = Rand (AST s) (E s ext t)
```

Comment:
Similarly, `Rand ext s t` seems like it would fit in with the existing conventions better than `Rand s ext t`.